src/main/java/com/gregtechceu/gtceu/api/mui/drawable/AdaptableUITexture.java

```diff
@@ -0,0 +1,201 @@
+            // center
+            GuiDraw.drawTiledTexture(pose, x + this.bl, y, width - this.bl - this.br, height, uInnerStart, this.v0,
+                    uInnerEnd, this.v1, tw - this.bl - this.br, th, 0);
+        } else if (this.bl <= 0 && this.br <= 0) {
+            // top border
+            GuiDraw.drawTiledTexture(pose, x, y, width, this.bt, this.u0, this.v0, this.u1, vInnerStart, tw, this.bt,
+                    0);
+            // bottom border
+            GuiDraw.drawTiledTexture(pose, x, y1 - this.bb, width, this.bb, this.u0, vInnerEnd, this.u1, this.v1, tw,
+                    this.bb, 0);
+            // center
+            GuiDraw.drawTiledTexture(pose, x, y + this.bt, width, height - this.bt - this.bb, this.u0, vInnerStart,
+                    this.u1, vInnerEnd, tw, th - this.bt - this.bb, 0);
+        } else {
+            // top left corner
+            GuiDraw.drawTiledTexture(pose, x, y, this.bl, this.bt, this.u0, this.v0, uInnerStart, vInnerStart, this.bl,
+                    this.bt, 0);
+            // top right corner
+            GuiDraw.drawTiledTexture(pose, x1 - this.br, y, this.br, this.bt, uInnerEnd, this.v0, this.u1, vInnerStart,
+                    this.br, this.bt, 0);
+            // bottom left corner
+            GuiDraw.drawTiledTexture(pose, x, y1 - this.bb, this.bl, this.bb, this.u0, vInnerEnd, uInnerStart, this.v1,
+                    this.bl, this.bb, 0);
+            // bottom right corner
+            GuiDraw.drawTiledTexture(pose, x1 - this.br, y1 - this.bb, this.br, this.bb, uInnerEnd, vInnerEnd, this.u1,
+                    this.v1, this.br, this.bb, 0);
+
+            // left border
+            GuiDraw.drawTiledTexture(pose, x, y + this.bt, this.bl, height - this.bt - this.bb, this.u0, vInnerStart,
+                    uInnerStart, vInnerEnd, this.bl, th - this.bt - this.bb, 0);
+            // top border
+            GuiDraw.drawTiledTexture(pose, x + this.bl, y, width - this.bl - this.br, this.bt, uInnerStart, this.v0,
+                    uInnerEnd, vInnerStart, tw - this.bl - this.bb, this.bt, 0);
+            // right border
+            GuiDraw.drawTiledTexture(pose, x1 - this.br, y + this.bt, this.br, height - this.bt - this.bb, uInnerEnd,
+                    vInnerStart, this.u1, vInnerEnd, this.br, th - this.bt - this.bb, 0);
+            // bottom border
+            GuiDraw.drawTiledTexture(pose, x + this.bl, y1 - this.bb, width - this.bl - this.br, this.bb, uInnerStart,
+                    vInnerEnd, uInnerEnd, this.v1, tw - this.bl - this.br, this.bb, 0);
+
+            // center
+            GuiDraw.drawTiledTexture(pose, x + this.bl, y + this.bt, width - this.bl - this.br,
+                    height - this.bt - this.bb, uInnerStart, vInnerStart, uInnerEnd, vInnerEnd,
+                    tw - this.bl - this.br, th - this.bt - this.bb, 0);
+        }
+        RenderSystem.disableBlend();
+    }
+
+    @Override
+    public boolean saveToJson(JsonObject json) {
```

Comment:
Doesn't this also need a loadFromJson?

Reply:
It doesn't. Thats handled in the super class. It makes more sense when you see the method